### PR TITLE
Ensure MP Rest Client providers can be registered via MP Config

### PR DESCRIPTION
**Pull request by andymc12**
Fixes and tests that when a user registers/prioritizes providers via MP Config that the MP Rest Client honors that. This resolves issue #5363.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Rebased with master and resolved conflicts.  New #build

File: dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.cdiPropsAndProviders/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+com.ibm.ws.logging.trace.specification=*=info:com.ibm.ws.jaxrs*=all:org.apache.cxf.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.cdiPropsAndProviders/bootstrap.properties]: we should limit logging to `*=info` unless you already suspect that there is a problem

File: dev/com.ibm.ws.microprofile.rest.client_fat/test-applications/cdiPropsAndProvidersApp/src/mpRestClient11/cdiPropsAndProviders/MyException.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public MyException() {
+        super();
+    }
+
+    /**
+     * @param message
+     */
+    public MyException(String message) {
+        super(message);
+        // TODO Auto-generated constructor stub
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.rest.client_fat/test-applications/cdiPropsAndProvidersApp/src/mpRestClient11/cdiPropsAndProviders/MyException.java]: I think we should remove these useless TODO statements

**Comment by andymc12** [comment]
Addressed code review changes - new #build

**Comment by andymc12** [comment]
Only failures (after only changing trace settings and removing comments) are frequently recurring intermittent test failures that are unrelated to these changes.  Merging.

**Pull request merged**